enterprise/server/remoteauth/remoteauth.go

```diff
@@ -0,0 +1,182 @@
+package remoteauth
+
+import (
+	"context"
+	"flag"
+	"net/http"
+
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/buildbuddy-io/buildbuddy/server/util/authutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/claims"
+	"github.com/buildbuddy-io/buildbuddy/server/util/grpc_client"
+	"github.com/buildbuddy-io/buildbuddy/server/util/hash"
+	"github.com/buildbuddy-io/buildbuddy/server/util/lru"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	authpb "github.com/buildbuddy-io/buildbuddy/proto/auth"
+)
+
+const (
+	jwtCacheSize = 10_000
+)
+
+var (
+	authHeaders = []string{authutil.APIKeyHeader}
+
+	remoteAuthTarget = flag.String("auth.remote_auth_target", "grpcs://remote.buildbuddy.dev", "The gRPC target of the remote authentication API.")
+)
+
+func NewRemoteAuthenticator() (*RemoteAuthenticator, error) {
+	conn, err := grpc_client.DialSimpleWithoutPooling(*remoteAuthTarget)
```

Comment:
pq no DialSimple?

Reply:
Just copied this from somewhere. Switched to DialSimple.